.changeset/tough-cars-yawn.md

```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes a bug where session data could be corrupted if it is changed after calling .set(); also fixes a bug where responses can be returned before session data is saved
```

Comment:
Can you split this into two chanegsets. You can run `pnpm changeset` again to add another one.